src/status_im/ui/screens/wallet/recipient/views.cljs

```diff
@@ -0,0 +1,286 @@
+(ns status-im.ui.screens.wallet.recipient.views
+  (:require [clojure.string :as string]
+            [re-frame.core :as re-frame]
+            [reagent.core :as reagent]
+            [status-im.i18n :as i18n]
+            [status-im.ui.components.react :as react]
+            [status-im.ui.components.topbar :as topbar]
+            [quo.core :as quo]
+            [status-im.ui.components.colors :as colors]
+            [status-im.utils.debounce :as debounce]
+            [status-im.ui.components.search-input.view :as search-input]
+            [status-im.ui.screens.wallet.components.views :as components]
+            [status-im.ui.components.icons.vector-icons :as icons]
+            [status-im.ui.components.chat-icon.screen :as chat-icon]
+            [status-im.multiaccounts.core :as multiaccounts]
+            [status-im.utils.utils :as utils]
+            [status-im.ui.components.keyboard-avoid-presentation :as kb-presentation]
+            [status-im.ui.components.toolbar :as toolbar]
+            [status-im.ethereum.core :as ethereum]
+            [status-im.ethereum.stateofus :as stateofus])
+  (:require-macros [status-im.utils.views :as views]))
+
+(defn- recipient-topbar []
+  [topbar/topbar
+   {:navigation    {:on-press
+                    #(do
+                       (re-frame/dispatch [:wallet/recipient-modal-closed])
+                       (re-frame/dispatch [:search/recipient-filter-changed nil])
+                       (re-frame/dispatch [:set-in [:wallet/prepare-transaction :modal-opened?] false])
```

Comment:
That concerns all `:set-in`'s, i won't repeat it in other places. I bet we discussed it many times already. imo that's not a big deal if we keep using it, but still a tiny dedicated handler instead would look abit better. Or if we can merge it into some neighbouring event.

Reply:
agree